llvm/lib/Target/RISCV/RISCVGatherScatterLowering.cpp

```diff
@@ -311,18 +307,31 @@ bool RISCVGatherScatterLowering::matchStridedRecurrence(Value *Index, Loop *L,
   }
   case Instruction::Mul: {
     Start = Builder.CreateMul(Start, SplatOp, "start");
-    Step = Builder.CreateMul(Step, SplatOp, "step");
```

Comment:
This block of code now entirely duplicates the end of matchStridedStart, maybe we can extract out a helper and use it in both places?  This an idea, not a requirement.

Reply:
Good catch. It looks like matchStridedStart doesn't set the IR names though and I don't want to introduce more potential diff, so I've left a todo to revisit this